pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -0,0 +1,283 @@
+			t.Errorf("Expected %t, got %t", tc.expected, got)
+		}
+	}
+}
+
+func TestIsHealthy(t *testing.T) {
+	nodeHealthy := node("healthy", true)
+	nodeUnhealthy := node("unhealthy", false)
+
+	if got := isHealthy(nodeHealthy); got != true {
+		t.Errorf("Expected true, got %t", got)
+	}
+	if got := isHealthy(nodeUnhealthy); got != false {
+		t.Errorf("Expected false, got %t", got)
+	}
+}
+
+func TestGetNodeCondition(t *testing.T) {
+	testsCases := []struct {
+		node      *corev1.Node
+		condition *corev1.NodeCondition
+		expected  *corev1.NodeCondition
+	}{
+		{
+			node: node("hasCondition", true),
+			condition: &corev1.NodeCondition{
+				Type:   corev1.NodeReady,
+				Status: corev1.ConditionTrue,
+			},
+			expected: &corev1.NodeCondition{
+				Type:   corev1.NodeReady,
+				Status: corev1.ConditionTrue,
+			},
+		},
+		{
+			node: node("doesNotHaveCondition", true),
+			condition: &corev1.NodeCondition{
+				Type:   corev1.NodeOutOfDisk,
+				Status: corev1.ConditionTrue,
+			},
+			expected: nil,
+		},
+	}
+
+	for _, tc := range testsCases {
+		got := getNodeCondition(tc.node, tc.condition.Type)
+		if !reflect.DeepEqual(got, tc.expected) {
+			t.Errorf("Expected %v, got %v", tc.expected, got)
+		}
+	}
+
+}
+
+type expectedReconcile struct {
+	result reconcile.Result
+	error  bool
+}
+
+func TestReconcile(t *testing.T) {
+	nodeHealthy := node("healthy", true)
+	nodeUnhealthy := node("unhealthy", false)
+	nodeWithNoMachineAnnotation := node("noAnnotated", true)
+	nodeWithNoMachineAnnotation.Annotations = map[string]string{}
+	nodeAnnotatedWithNoExistentMachine := node("noExistentMachine", true)
+	nodeAnnotatedWithNoExistentMachine.Annotations[machineAnnotationKey] = "noExistentMachine"
+	fakeMachine := machine("fakeMachine")
+
+	testsCases := []struct {
+		node     *v1.Node
+		expected expectedReconcile
+	}{
+		{
+			node: nodeHealthy,
+			expected: expectedReconcile{
+				result: reconcile.Result{},
+				error:  false,
+			},
+		},
+		{
+			node: nodeUnhealthy,
+			expected: expectedReconcile{
+				result: reconcile.Result{},
+				error:  false,
+			},
+		},
+		{
+			node: nodeWithNoMachineAnnotation,
+			expected: expectedReconcile{
+				result: reconcile.Result{},
+				error:  false,
+			},
+		},
+		{
+			node: nodeAnnotatedWithNoExistentMachine,
+			expected: expectedReconcile{
+				result: reconcile.Result{},
+				error:  false,
+			},
+		},
+	}
+
+	machineHealthCheck := machineHealthCheck("machineHealthCheck")
+	allMachineHealthChecks := &healthcheckingv1alpha1.MachineHealthCheckList{
+		Items: []healthcheckingv1alpha1.MachineHealthCheck{
+			*machineHealthCheck,
+		},
+	}
+
+	r := newFakeReconciler(nodeHealthy, nodeUnhealthy, fakeMachine, allMachineHealthChecks)
+	for _, tc := range testsCases {
+		request := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: "",
+				Name:      tc.node.Name,
+			},
+		}
+		result, err := r.Reconcile(request)
+		if result != tc.expected.result {
```

Comment:
Should this not check `err` first, before asserting on `result`?

Reply:
not really, result and err are the tuple expectation